src/main/java/com/enderio/base/common/enchantment/AutoSmeltModifier.java

```diff
@@ -0,0 +1,45 @@
+package com.enderio.base.common.enchantment;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
+import net.minecraftforge.common.loot.IGlobalLootModifier;
+import net.minecraftforge.common.loot.LootModifier;
+import net.minecraftforge.items.ItemHandlerHelper;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.stream.Collectors;
+
+// Based on forge example: https://github.com/MinecraftForge/MinecraftForge/blob/1.20.x/src/test/java/net/minecraftforge/debug/gameplay/loot/GlobalLootModifiersTest.java#L137
+public class AutoSmeltModifier extends LootModifier {
+    public static final Supplier<Codec<AutoSmeltModifier>> CODEC = Suppliers.memoize(() -> RecordCodecBuilder.create(inst -> codecStart(inst).apply(inst, AutoSmeltModifier::new)));
+
+    public AutoSmeltModifier(LootItemCondition[] conditionsIn) {
+        super(conditionsIn);
+    }
+
+    @Override
+    protected @NotNull ObjectArrayList<ItemStack> doApply(ObjectArrayList<ItemStack> generatedLoot, LootContext context) {
+        var level = context.getLevel();
+        var recipeManager = level.getRecipeManager();
+        return generatedLoot.stream().map(stack ->
```

Comment:
I think this is fine, just know we are smelting the drops, not the block itself (not sure this was different before). This means if you mine ore blocks you get the smelted chunks, not the smelted ore block (so more than just smelting the block).

Reply:
Yeah that'd be the intention :)